File: backend/TenderGo.Models/Requests/TenderInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace TenderGo.Models.Requests
{
    public class TenderInsertRequest
    {
        [Required(ErrorMessage = "Title is required.")]
        [MaxLength(200, ErrorMessage = "Title cannot exceed 200 characters.")]
        public string Title { get; set; } = string.Empty;

        [Required(ErrorMessage = "Max budget is required.")]
        [Range(0.01, double.MaxValue, ErrorMessage = "MaxBudget must be a positive number greater than 0.")]
        public decimal MaxBudget { get; set; }

        [Required(ErrorMessage = "Location is required.")]
        public int LocationId { get; set; }

        [MaxLength(1000, ErrorMessage = "Description cannot exceed 1000 characters.")]
        public string? Description { get; set; }

        [Required(ErrorMessage = "Category is required.")]
        public int CategoryId { get; set; }

        [Required(ErrorMessage = "Deadline is required.")]
        public DateTime Deadline { get; set; }
        public List<byte[]>? ImageBytes { get; set; } = new();
    }
}

File: UI/tendergo_mobile/lib/shared/models/requests/tender_insert_request.dart
import 'package:tendergo/shared/models/dto/tender_image_dto.dart';

class TenderInsertRequest {
  final String title;
  final double maxBudget; 
  final int locationId;
  final String? description; 
  final int categoryId;
  final DateTime deadline;
  final List<TenderImageDto> images;

  TenderInsertRequest({
    required this.title,
    required this.maxBudget,
    required this.locationId,
    this.description,
    required this.categoryId,
    required this.deadline,
    this.images = const [],
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'maxBudget': maxBudget, 
      'locationId': locationId,
      'description': description, 
      'categoryId': categoryId,
      'deadline': deadline.toIso8601String(),
      'images': images
          .map(
            (image) => {
              'imageUrl': image.imageUrl,
              'imageHash': image.imageHash,
            },
          )
          .toList(),
    };
  }
}